crates/relayer-types/src/core/ics04_channel/channel.rs

```diff
@@ -460,20 +459,30 @@ impl State {
             0 => Ok(Self::Uninitialized),
             1 => Ok(Self::Init),
             2 => Ok(Self::TryOpen),
-            3 => Ok(Self::Open),
+            3 => Ok(Self::Open(UpgradeState::NotUpgrading)),
             4 => Ok(Self::Closed),
             5 => Ok(Self::Flushing),
             6 => Ok(Self::Flushcomplete),
-            7 => Ok(Self::InitUpgrade),
-            8 => Ok(Self::TryUpgrade),
-            9 => Ok(Self::AckUpgrade),
             _ => Err(Error::unknown_state(s)),
         }
     }
 
+    // Parses the State out from a i32.
+    pub fn into_i32(&self) -> i32 {
+        match self {
+            State::Uninitialized => 0,
+            State::Init => 1,
+            State::TryOpen => 2,
+            State::Open(_) => 3,
+            State::Closed => 4,
+            State::Flushing => 5,
+            State::Flushcomplete => 6,
+        }
+    }
```

Comment:
Could we instead implement this conversion using the `From` trait? I.e.
```rust
impl From<State> for i32 {
    fn from(state: State) -> Self {
        match self {
            State::Uninitialized => 0,
            State::Init => 1,
            State::TryOpen => 2,
            State::Open(_) => 3,
            State::Closed => 4,
            State::Flushing => 5,
            State::Flushcomplete => 6,
        }
    }
}
```

Reply:
I'd rather keep it explicit, From does not really buy us anything here (`.into_i32()` vs `.into()`) since we are not using this State generically. I think we've (and I include myself here ofc) abused a bit of From instances over the years and should from now on try to keep things more explicit unless we have a concrete need for a specific instance.

Reply:
I would rename the method to `as_i32` for consistency with `as i32` and the general naming conventions in Rust:

- `as_X()`: `&self -> X`
- `into_X()`: `self -> X`